velodyne_pointcloud/src/conversions/convert.cc

```diff
@@ -21,29 +21,36 @@ namespace velodyne_pointcloud
 {
   /** @brief Constructor. */
   Convert::Convert(ros::NodeHandle node, ros::NodeHandle private_nh):
-    data_(new velodyne_rawdata::RawData())
+    data_(new velodyne_rawdata::RawData()),
+    prev_azimuth_(0.0),
+    prev_stamp_(ros::Time())
   {
     data_->setup(private_nh);
 
 
     // advertise output point cloud (before subscribing to input data)
-    output_ =
+    output_pointcloud_ =
       node.advertise<sensor_msgs::PointCloud2>("velodyne_points", 10);
```

Comment:
do we need a queue size this bit? Is this the whole point cloud?

Reply:
big*

Reply:
Yes, this is for the whole pointcloud. This was already 10 to begin with. Have reduced it to 2 and works fine when I tested.